File: aricode/sparring/challenges/08_gcd/gcd.go
// Go - Challenge 08: Euclidean GCD
// Compute GCD(462, 1071) = 21
package main

import "os"

func gcd(a, b int) int {
	for b > 0 {
		a, b = b, a%b
	}
	return a
}

func main() {
	os.Exit(gcd(462, 1071))
}

File: aricode/sparring/challenges/08_gcd/gcd.py
#!/usr/bin/env python3
# Python - Challenge 08: Euclidean GCD
# Compute GCD(462, 1071) = 21
import sys

def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a

sys.exit(gcd(462, 1071))
